compat/src/index.d.ts

```diff
@@ -0,0 +1,71 @@
+import * as hooks from '../../hooks';
+import { VNode, PreactElement } from '../../src/internal';
+import { Component, createContext, createRef, Fragment, toChildArray, ComponentChildren, ComponentChild } from '../../src';
+import { FunctionalComponent } from 'preact';
+import { ForwardFn } from './internal';
+
+export { Component, createContext, createRef, Fragment } from '../../src';
+export * from '../../hooks';
+
+export const version: string;
+
+export function createPortal (vnode: VNode, container: PreactElement): VNode<any>;
+
+export function createElement (type: VNode["type"], props?: any, Children?: ComponentChildren): VNode<any>;
+
+export function cloneElement (element: VNode<any>): VNode<any>;
+
+export function render (vnode: VNode<any>, parent: PreactElement, callback?: () => void): Component | null;
+
+export function unmountComponentAtNode (container: Element | Document | ShadowRoot | DocumentFragment): boolean;
+
+export function createFactory (type: VNode["type"]): VNode<any>;
+
+export function isValidElement (element: any): boolean;
+
+export function findDOMNode (component: Component): PreactElement | null;
+
+export class PureComponent extends Component<P, S> {
+  public isPureReactComponenet: boolean;
+
+  public shouldComponentUpdate (props: P, state: S): boolean;
+}
+
+export function memo (c: FunctionalComponent, comparer: (prev: any, next: any) => boolean): FunctionalComponent;
+
+export function forwardRef (fn: ForwardFn): FunctionalComponent;
+
+export function unstable_batchedUpdates (callback: () => void, arg?: any): void;
+
+function mapFn (children: ComponentChildren, fn: (d: any, i: number) => any): Array<VNode | null> | null;
+
+export interface Children {
+  map: mapFn;
+  forEach: mapFn;
```

Comment:
Although both `map` and `forEach` share the same implementation internally for byte saving reasons, we should not make that public.

```ts
export interface Children {
  map<T extends ComponentChild, R>(children: T | T[], fn: (child: T, i: number, array: T[]) => R): R[];
  forEach<T extends ComponentChild, R>(children: T | T[], fn: (child: T, i: number, array: T[]) => void): void;
  // ...
}
```

Reply:
Updated.